backend/src/apiserver/storage/db.go

```diff
@@ -59,6 +61,9 @@ type SQLDialect interface {
 
 	// Inserts new rows and updates duplicates based on the key column.
 	Upsert(query string, key string, overwrite bool, columns ...string) string
+
+	// Updates a run_details based on join (production) or inner select (test).
+	UpdateRunDetailsStorageState(expId string) (string, []interface{})
```

Comment:
Hi @difince, thank you so much for your help in improving the query performance! It looks like the existing methods in this interface `SQLDialect` are mostly common database operations (except `IsDuplicateError()`). Can we do something similar here? We can use [`UpdateJoin()`](https://stackoverflow.com/questions/19270259/update-with-join-in-sqlite) or [`UpdateFrom()`](https://www.sqlite.org/lang_update.html#upfrom). Also according to [here](https://stackoverflow.com/a/69831549/8216546), SQLite supports UPDATE FROM starting v3.33. If we are using a version later than that, we don't have to use nested selects with SQLite either.

Reply:
Then maybe it would make sense to improve all database queries. We are also experience heavy slowdowns with the listruns database query.

Reply:
Hello @Linchin Thank you for your review and comments. 
I rewrote the SQLite request to use the UPDATE-FROM construct. This made the code a bit clearer, and I was able to abstract away (at some degree) the implementation within `SQLDialect` - trying to make it a bit more like a `common database operation`

I was unable to make the SQLite works with UpdateJoin. Most of the examples I browsed were using inner select, including the one referenced in this link - [UpdateJoin](https://stackoverflow.com/questions/19270259/update-with-join-in-sqlite). I have checked the SQLite version we work with - `3.39.4`
Still, if we take a closer look at the diagram of SQLite's official documentation about [Update Statement ](https://www.sqlite.org/lang_update.html)- and goes deeper in `join-clause` and `join-operator` - it looks like there is some support - but I really didn't make it work and didn't find any reference in the internet. 

Still, my last commit improves the code.

Reply:
Here are the current versions of the requests, if this will make the review easier
 
**MySQL**:
`UPDATE run_details JOIN resource_references ON run_details.UUID = resource_references.ResourceUUID SET StorageState = "STORAGESTATE_ARCHIVED" WHERE resource_references.ResourceType = "RUN" AND resource_references.ReferenceUUID = "<experiment_id>" AND resource_references.ReferenceType = "Experiment"
`

**SQLite**:
`UPDATE run_details SET StorageState = "STORAGESTATE_ARCHIVED" FROM resource_references WHERE run_details.UUID = resource_references.ResourceUUID AND resource_references.ResourceType = "RUN" AND resource_references.ReferenceUUID = "<experiment_id>" AND resource_references.ReferenceType = "Experiment"`

Reply:
@juliusvonkohout thank you for bringing this up. Let's keep the scope of this PR to what it is now. Would you create an issue on the listruns query speed, and if there are any other operations with similar speed issue, add them too?